spring-cloud-stream-binders/spring-cloud-stream-binder-gemfire/src/test/java/org/springframework/cloud/stream/binder/gemfire/GemfireBinderTests.java

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder.gemfire;
+
+import static org.junit.Assert.*;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+
+import com.gemstone.gemfire.distributed.LocatorLauncher;
+import com.oracle.tools.runtime.LocalPlatform;
+import com.oracle.tools.runtime.PropertiesBuilder;
+import com.oracle.tools.runtime.concurrent.RemoteCallable;
+import com.oracle.tools.runtime.console.SystemApplicationConsole;
+import com.oracle.tools.runtime.java.JavaApplication;
+import com.oracle.tools.runtime.java.LocalJavaApplicationBuilder;
+import com.oracle.tools.runtime.java.SimpleJavaApplication;
+import com.oracle.tools.runtime.java.SimpleJavaApplicationSchema;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.MessagingException;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.messaging.support.ExecutorSubscribableChannel;
+import org.springframework.messaging.support.GenericMessage;
+
+/**
+ * Tests for {@link GemfireMessageChannelBinder}.
+ *
+ * @author Patrick Peralta
+ */
+public class GemfireBinderTests {
+	private static final Logger logger = LoggerFactory.getLogger(GemfireBinderTests.class);
+
+	/**
+	 * Timeout value in milliseconds for operations to complete.
+	 */
+	private static final long TIMEOUT = 30000;
+
+	/**
+	 * Port number for Gemfire locator.
+	 */
+	private static final int LOCATOR_PORT = 7777;
+
+	/**
+	 * Payload of test message.
+	 */
+	private static final String MESSAGE_PAYLOAD = "hello world";
+
+	/**
+	 * Name of binding used for producer and consumer bindings.
+	 */
+	private static final String BINDING_NAME = "test";
+
+	/**
+	 * Test basic message sending functionality.
+	 *
+	 * @throws Exception
+	 */
+	@Test
+	public void testMessageSendReceive() throws Exception {
+		LocatorLauncher locatorLauncher = null;
+		JavaApplication consumer = null;
+		JavaApplication producer = null;
+
+		try {
+			locatorLauncher = new LocatorLauncher.Builder()
```

Comment:
Yup, we can definitely extract this test code at some point. I purposefully didn't do it as part of this PR. As I add more tests for different parts of the functionality, it will be easier to determine the correct abstractions for GF tests.